value.go

```diff
@@ -196,3 +208,39 @@ func (BinaryValueCodec) UnmarshalValueKeys(b []byte) ([][]byte, error) {
 	}
 	return vk, nil
 }
+
+func (bjc BinaryWithJsonFallbackCodec) MarshalValue(v Value) ([]byte, error) {
+	return bjc.BinaryValueCodec.MarshalValue(v)
+}
+
+func (bjc BinaryWithJsonFallbackCodec) UnmarshalValue(b []byte) (Value, error) {
+	// Use a quick check to probe if b is encoded in JSON format.
+	// Expect a JSON object.
+	if len(b) > 0 && b[0] == '{' {
+		// Attempt to decode value as JSON and ultimately fall back on binary if JSON decode fails
+		v, err := bjc.JsonValueCodec.UnmarshalValue(b)
+		if err == nil {
+			return v, nil
+		}
+		log.Warn("Unmarshalling value as JSON failed; falling back on binary codec.", "err", err)
```

Comment:
this probably isn't a warning - as the datastore moves to mostly binary, you'll still end up with 1/256 values being binary but having a trial-json-decrypt that triggers this. probably can quiet it